source/RaidFleet.cpp

```diff
@@ -0,0 +1,64 @@
+/* RaidFleet.cpp
+Copyright (c) 2023 by Hurleveur
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "RaidFleet.h"
+
+#include "DataNode.h"
+#include "Fleet.h"
+#include "GameData.h"
+
+#include <algorithm>
+
+
+RaidFleet::RaidFleet(const Fleet *fleet, double minAttraction, double maxAttraction)
+	: fleet(fleet), minAttraction(minAttraction), maxAttraction(maxAttraction)
+{
+}
+
+
+
+void RaidFleet::Load(const DataNode &node, std::vector<RaidFleet> &raidFleets, bool remove, int valueIndex)
+{
+	const Fleet *fleet = GameData::Fleets().Get(node.Token(valueIndex));
+	if(remove)
+		raidFleets.erase(remove_if(raidFleets.begin(), raidFleets.end(),
+			[fleet](const RaidFleet &raidFleet) noexcept -> bool { return raidFleet.GetFleet() == fleet; }),
+		raidFleets.end());
```

Comment:
This feels a bit overly cluttered.
```suggestion
	{
		auto fleetMatcher = [fleet](const RaidFleet &raidFleet) noexcept -> bool {
			return raidFleet.GetFleet() == fleet;
		};
		raidFleets.erase(remove_if(raidFleets.begin(), raidFleets.end(), fleetMatcher), raidFleets.end());
	}
```